tests/functional/src/main/java/org/mule/functional/junit4/runners/ArtifactClassLoaderRunnerConfig.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.functional.junit4.runners;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Specifies a configuration needed by {@link ArtifactClassloaderTestRunner} in order to
+ * run the test.
+ *
+ * @since 4.0
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+@Inherited
+public @interface ArtifactClassLoaderRunnerConfig
+{
+
+    /**
+     * @return a comma separated list of packages to be added as PARENT_ONLY for the
+     * container class loader, default (and required) packages is empty.
+     * In case of having to append one you should also include the default list.
+     */
+    String extraBootPackages() default "";
+
+    /**
+     * @return {@link String[]} with the extension {@link Class} name of the extensions that would be used to create and load it using a plugin/extensio
+     * {@link ClassLoader}. If no extensions are defined the plugin/extension class loaders will not be created.
+     */
+    String[] extensions() default {};
+
+    /**
+     * @return a comma separated list of groupId:artifactId:type (it does support wildcards org.mule:*:* or *:mule-core:* but
+     * only starts with for partial matching org.mule*:*:*) that would be used in order to exclude artifacts that should not be added to
+     * the application class loader neither the extension/plugin class loaders due to they will be already exposed through the container.
+     * Default exclusion is already defined in excluded.properties file and by using this annotation the ones defined here will be appended
+     * to those defined in file.
+     */
+    String exclusions() default "";
```

Comment:
Let's review this together, next commit will have a new entity MuleArtifactFunctionalTestCase that will define the default values (exclusions, basePackage for extensions, etc) that would be used by Mule components to test features in an isolated environment and the excluded.properties will only have junit and hamcrest that must go to container always.